files/default/neutron-evacuate-lbaasv2-agent.py

```diff
@@ -0,0 +1,223 @@
+            "WHERE agents.host = %s "
+            "AND agents.binary='neutron-lbaasv2-agent' "
+            "AND agents.id=lbaas_loadbalanceragentbindings.agent_id;",
+            agent_hostname)
+        return [id[0] for id in ret.fetchall()]
+
+    def reassign_loadbalancers(self, load_balancer_ids, target_agents):
+        for agent in target_agents:
+            agent['load'] = len(self.loadbalancers_on_agent(agent['host']))
+
+        agents_to_restart = set()
+        for loadbalancer in load_balancer_ids:
+            agent = min(target_agents, key=lambda x: x['load'])
+            LOG.info("Reassigning loadbalancer %s to agent %s" %
+                     (loadbalancer, agent))
+
+            self.connection.execute(
+                "UPDATE lbaas_loadbalanceragentbindings "
+                "SET agent_id = %s "
+                "WHERE loadbalancer_id = %s;",
+                agent['id'], loadbalancer
+            )
+            agents_to_restart.add(agent['host'])
+            agent['load'] = agent['load'] + 1
+        return agents_to_restart
+
+    def setup(self):
+        cfg.CONF(project='neutron')
+        # Always log to stderr, ignore setting in neutron.conf
+        cfg.CONF.set_override("use_stderr", True)
+        logging.setup(cfg.CONF, "neutron")
+        self.host_to_evacuate = cfg.CONF.host
+
+        db_uri = cfg.CONF.database.connection
+        db = sqlalchemy.create_engine(db_uri)
+        try:
+            self.connection = db.connect()
+        except sqlalchemy.exc.SQLAlchemyError as e:
+            LOG.error('Cannot connect to database: %s' % e)
+            raise
+
+    def run(self):
+        load_balancers = self.loadbalancers_on_agent(self.host_to_evacuate)
+        if load_balancers:
+            LOG.info("Evacuating loadbalancers: %s", load_balancers)
+            target_agents = self.available_destination_agents()
+            LOG.info("Available destination agents %s", target_agents)
+            agents_to_restart = list(
+                self.reassign_loadbalancers(load_balancers, target_agents)
+            )
+
+            if cfg.CONF.source_agent_restart:
+                # Make sure the source agent is restarted first
+                agents_to_restart.insert(0, self.host_to_evacuate)
+
+            LOG.info("agents to restart: %s" % agents_to_restart)
+
+            for host in agents_to_restart:
+                LOG.info("restarting agent on %s" % host)
+                cleanup = RemoteLbaasV2Cleanup(host, timeout=30)
+                if (host != self.host_to_evacuate or
+                    cfg.CONF.source_agent_restart):
+                    if cfg.CONF.use_crm:
+                        cleanup.restart_lbaasv2_agent_crm()
+                    else:
+                        cleanup.restart_lbaasv2_agent_systemd()
+
+                if (host == self.host_to_evacuate and
+                    cfg.CONF.delete_namespaces):
+                    cleanup.delete_lbaasv2_namesspaces(load_balancers)
```

Comment:
Actually there is still a bug which prevents " cfg.CONF.delete_namespaces" from working correctly. I though I already addressed that :/